packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/sagemaker/create-training-job.ts

```diff
@@ -163,6 +163,14 @@ export class SageMakerCreateTrainingJob extends sfn.TaskStateBase implements iam
       throw new Error('Must define either an algorithm name or training image URI in the algorithm specification');
     }
 
+    // validate the algorithmName if the algorithmName is defined
+    if (props.algorithmSpecification.algorithmName) {
```

Comment:
I think we should create a private `validateAlgorithmName` function to do the validation which:
1. Returns if `algorithmName` is `undefined`
2. Throws if the length of `algorithmName` is [not between 1 and 170](https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_AlgorithmSpecification.html#:~:text=Length%20Constraints%3A%20Minimum%20length%20of%201.%20Maximum%20length%20of%20170.)
3. Throws if `algorithmName` doesn't satisfy the regex pattern

Reply:
I just missed 2. Thank you for pointing it out.
I will implement it as you suggested.